src/fping.c

```diff
@@ -2939,6 +2955,34 @@ void ev_remove(struct event_queue *queue, struct event *event)
     event->ev_next = NULL;
 }
 
+/************************************************************
+
+  Function: print_human_readable_time from current_time_ns
+
+*************************************************************/
+void print_timestamp_format(int64_t current_time_ns, int timestamp_format)
+{
+    char time_buffer[100];
+    time_t current_time_s;
+    struct tm *local_time;
+
+    if(timestamp_format >= 1 && timestamp_format <= 3) {
+        current_time_s = current_time_ns / 1000000000;
+        local_time = localtime(&current_time_s);
+        if(timestamp_format == 1) {
```

Comment:
I'd think this part would be more readable using `switch`.

Reply:
The if statement has been completely converted into a switch statement. Many thanks for this feedback.
Switch statement with the default is also better than the old if else.
This way, the default is always used if the function is passed incorrectly